Runtime/safe-ds/safe_ds/data/_ordinal_encoder.py

```diff
@@ -0,0 +1,132 @@
+from safe_ds import exceptions
+from safe_ds.data._table import Table
+from sklearn import preprocessing
+
+
+# noinspection PyProtectedMember
+class OrdinalEncoder:
+    """
+    This OrdinalEncoder encodes one or more given columns into ordinal numbers. The encoding order must be provided.
+
+    Parameters
+    --------
+        order : list[str]
+            The order in which the ordinal encoder encodes the values
+    """
+
+    def __init__(self, order: list[str]) -> None:
+        self.is_fitted = 0
+        self.oe = preprocessing.OrdinalEncoder(categories=[order])
+        self.order = order
+
+    def fit(self, table: Table, column_name: str) -> None:
+        """
+        Fit the ordinal encoder with the values in the given table.
+
+        Parameters
+        ----------
+        table : Table
+            The table containing the data to fit the ordinal encoder with.
+
+        column_name : str
+            The column which should be ordinal-encoded
+
+        Returns
+        -------
+        None
+            This function does not return any value. It updates the internal state of the ordinal encoder object.
+
+        Raises
+        -------
+            LearningError if the Model couldn't be fitted correctly
+        """
+
+        p_df = table._data
+        p_df.columns = table.schema.get_column_names()
+        try:
+            self.oe.fit(p_df[[column_name]])
+        except exceptions.NotFittedError as exc:
+            raise exceptions.LearningError("") from exc
+
+    def transform(self, table: Table, column_name: str) -> Table:
+        """
+        Transform the given Table to a ordinal-encoded table.
+
+        Parameters
+         ----------
+         table:
+                 table with target values
+         column_name:
+                 name of column as string
+         Returns
+         -------
+             Table with ordinal encodings.
+
+         Raises
+         ------
+             a NotFittedError if the Model wasn't fitted before transforming
+        """
+        p_df = table._data.copy()
+        p_df.columns = table.schema.get_column_names()
+        try:
+            p_df[[column_name]] = self.oe.transform(p_df[[column_name]])
+            p_df[column_name] = p_df[column_name].astype(dtype="int64", copy=False)
+            return Table(p_df)
```

Comment:
```suggestion
            return Table(p_df, table.schema)
```
Otherwise columnnames wouldn't be ratained, no?